ingest/src/main/java/com/microsoft/azure/kusto/ingest/IngestClientImpl.java

```diff
@@ -144,27 +160,34 @@ public IngestionResult ingestFromStream(StreamSourceInfo streamSourceInfo, Inges
                 streamSourceInfo.getStream().close();
             }
             return ingestionResult;
-        } catch (Exception ex) {
-            log.error("ingestFromStream: Error while ingesting from stream.", ex);
-            throw ex;
+        } catch (IOException | URISyntaxException e) {
+            throw new IngestionClientException("Error in ingestFromStream()", e);
+        } catch (StorageException e) {
+            throw new IngestionServiceException("Error in ingestFromStream()", e);
         }
     }
 
-    private Long estimateBlobRawSize(@org.jetbrains.annotations.NotNull BlobSourceInfo blobSourceInfo) throws Exception {
-        String blobPath = blobSourceInfo.getBlobPath();
-        CloudBlockBlob blockBlob = new CloudBlockBlob(new URI(blobPath));
-        blockBlob.downloadAttributes();
-        long length = blockBlob.getProperties().getLength();
+    private Long estimateBlobRawSize(@org.jetbrains.annotations.NotNull BlobSourceInfo blobSourceInfo) throws IngestionClientException, IngestionServiceException {
+        try {
+            String blobPath = blobSourceInfo.getBlobPath();
+            CloudBlockBlob blockBlob = new CloudBlockBlob(new URI(blobPath));
+            blockBlob.downloadAttributes();
+            long length = blockBlob.getProperties().getLength();
 
-        if (length == 0) {
-            return length;
-        }
+            if (length == 0) {
+                return length;
+            }
 
-        if (blobPath.contains(".zip") || blobPath.contains(".gz")) {
-            length = length * COMPRESSED_FILE_MULTIPLIER;
-        }
+            if (blobPath.contains(".zip") || blobPath.contains(".gz")) {
+                length = length * COMPRESSED_FILE_MULTIPLIER;
+            }
 
-        return length;
+            return length;
+        } catch (StorageException e) {
```

Comment:
IIUC, those excpetions are caught in ingestFromFile, do we have any value in catching them twice?

Reply:
We call this method (estimateBlobRawSize()) from ingestFromBlob().
You are right, we can just throw them and catch in ingestFromBlob().